src/lib/reasoners/satml_frontend.ml

```diff
@@ -1035,6 +1038,70 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
           env i ~unknown_reason:(Timeout ProofSearch) (* may becomes ModelGen *)
       end
 
+
+  let [@inline always] gt a b = E.mk_builtin ~is_pos:false Symbols.LE [a;b]
+  let [@inline always] lt a b = E.mk_builtin ~is_pos:true  Symbols.LT [a;b]
+
+  let analyze_unknown_for_objectives env unknown_exn unsat_rec_prem : unit =
+    let obj = Th.get_objectives (SAT.current_tbox env.satml) in
+    if Util.MI.is_empty obj then raise unknown_exn;
+    env.objectives := Some obj;
+    let seen_infinity = ref false in
+    let acc =
+      Util.MI.fold
+        (fun _ {Th_util.e; value; r=_; is_max; order=_} acc ->
+           if !seen_infinity then acc
+           else
+             match value with
+             | Pinfinity ->
+               seen_infinity := true; acc
+             | Minfinity ->
+               seen_infinity := true; acc
+             | Value (_,_, Th_util.CS (Some {opt_val = Value v; _},_,_)) ->
+               (* hack-ish to get the value of type expr *)
+               (e, v, is_max) :: acc
+             | Value _ ->
+               assert false
+             | Unknown ->
+               seen_infinity := true; acc
```

Comment:
This doesn't seem correct. `Unknown` means `Unknown`, not `infinity`.

Reply:
See other comment above — if we use an explicit `StrictBound` as suggested above, we can get rid of this. Note that `StrictBound` needs to contain the value of the bound to be updated in the "should continue beyond SAT" below.

Reply:
I used an explicit `StrictBound` in this very location. I don't know why I modified this line. I revert the modification.